src/vs/platform/actionWidget/browser/actionWidget.ts

```diff
@@ -0,0 +1,625 @@
+
+export const acceptSelectedAction = 'acceptSelectedAction';
+export const previewSelectedAction = 'previewSelectedAction';
+
+export const Context = {
+	Visible: new RawContextKey<boolean>('actionWidgetVisible', false, localize('actionWidgetVisible', "Whether the action widget list is visible"))
+};
+export interface IRenderDelegate<T> {
+	onHide(cancelled: boolean): void;
+	onSelect(action: T, trigger: any, preview?: boolean): Promise<any>;
+}
+
+export interface IActionShowOptions {
+	readonly includeDisabledActions: boolean;
+	readonly fromLightbulb?: boolean;
+	readonly showHeaders?: boolean;
+}
+
+export interface IListMenuItem<T extends IActionItem> {
+	item?: T;
+	kind?: any;
+	group?: any;
+	disabled?: boolean;
+	label?: string;
+}
+
+export interface IActionList<T extends IActionItem> extends IDisposable {
+	hide(): void;
+	focusPrevious(): void;
+	focusNext(): void;
+	layout(minWidth: number): void;
+	toMenuItems(items: readonly T[], showHeaders: boolean): IListMenuItem<T>[];
+	acceptSelected(preview?: boolean): void;
+	domNode: HTMLElement;
+}
+
+export interface IActionMenuTemplateData {
+	readonly container: HTMLElement;
+	readonly icon: HTMLElement;
+	readonly text: HTMLElement;
+	readonly keybinding: KeybindingLabel;
+}
+
+export const enum ActionListItemKind {
+	Action = 'action',
+	Header = 'header'
+}
+
+interface IHeaderTemplateData {
+	readonly container: HTMLElement;
+	readonly text: HTMLElement;
+}
+
+export class HeaderRenderer<T extends IListMenuItem<IActionItem>> implements IListRenderer<T, IHeaderTemplateData> {
+
+	get templateId(): string { return ActionListItemKind.Header; }
+
+	renderTemplate(container: HTMLElement): IHeaderTemplateData {
+		container.classList.add('group-header');
+
+		const text = document.createElement('span');
+		container.append(text);
+
+		return { container, text };
+	}
+
+	renderElement(element: IListMenuItem<IActionItem>, _index: number, templateData: IHeaderTemplateData): void {
+		templateData.text.textContent = element.group.title;
+	}
+
+	disposeTemplate(_templateData: IHeaderTemplateData): void {
+		// noop
+	}
+}
+
+
+export class ActionItemRenderer<T extends IListMenuItem<IActionItem>> implements IListRenderer<T, IActionMenuTemplateData> {
+
+	get templateId(): string { return 'action'; }
+
+	constructor(private readonly _keybindingResolver: any, @IKeybindingService private readonly _keybindingService: IKeybindingService) {
+	}
+
+	renderTemplate(container: HTMLElement): IActionMenuTemplateData {
+		container.classList.add(this.templateId);
+
+		const icon = document.createElement('div');
+		icon.className = 'icon';
+		container.append(icon);
+
+		const text = document.createElement('span');
+		text.className = 'title';
+		container.append(text);
+
+		const keybinding = new KeybindingLabel(container, OS);
+
+		return { container, icon, text, keybinding };
+	}
+
+	renderElement(element: T, _index: number, data: IActionMenuTemplateData): void {
```

Comment:
nit: never seen a _ _prefix_ to indicate an unused arg before

Reply:
it avoids the lint error to have the prefix